apps/web/src/hooks/useWorkflowChannelPreferences.ts

```diff
@@ -0,0 +1,74 @@
+import { ChannelTypeEnum, IResponseError } from '@novu/shared';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { QueryKeys } from '../api/query.keys';
+import { useDiscover } from '../studio/hooks/useBridgeAPI';
+import { useStudioState } from '../studio/StudioStateProvider';
+import { errorMessage, successMessage } from '../utils/notifications';
+import { useNovuAPI } from './useNovuAPI';
+
+type ChannelPreference = {
+  defaultValue: boolean;
+  readOnly: boolean;
+};
+
+export type WorkflowChannelPreferences = {
+  workflow: ChannelPreference;
+  channels: {
+    [key in (typeof ChannelTypeEnum)[keyof typeof ChannelTypeEnum]]: ChannelPreference;
+  };
+};
+
+export const useWorkflowChannelPreferences = (
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
💭 **thought (non-blocking):** While I realize the ticket says "create _a‏_ hook", I don't think it's beneficial to put all of this behavior into a single hook. I think these could be three separate hooks, which can then be better used in their respective contexts: 

1. `useUpdateWorkflowChannelPreferences` - Handles only the upsert functionality. I do think we should use `update` because in reality, we already have default preferences available, so it is really "updating" from most perspectives
2. `useStudioWorkflowChannelPreferences` - Gets preferences only for local studio.
3. `useCloudWorkflowChannelPreferences` - Gets preferences only for the cloud.

I don't think it makes sense to merge the two "getters" (2 & 3) when they'll be used in different components accordingly. By separating the `update` out, we can compose the hooks however we want without bringing in unnecessary re-renders